packages/progressive-profiling-nodejs/src/implementation.ts

```diff
@@ -0,0 +1,314 @@
+import {
+  RegisterSections,
+  FormSection,
+  SuperTokensPluginProfileProgressiveProfilingNormalisedConfig,
+  UserMetadataConfig,
+} from "./types";
+import { logDebugMessage } from "./logger";
+import { FormField, FormFieldValue, ProfileFormData } from "@supertokens-plugins/progressive-profiling-shared";
+import { SessionContainerInterface } from "supertokens-node/recipe/session/types";
+import { BooleanClaim } from "supertokens-node/recipe/session/claims";
+import { PLUGIN_ID, METADATA_KEY, METADATA_PROFILE_KEY } from "./constants";
+import { pluginUserMetadata } from "@shared/nodejs";
+import { BasePluginImplementation, groupBy, indexBy, mapBy } from "@shared/js";
+
+export class Implementation extends BasePluginImplementation<SuperTokensPluginProfileProgressiveProfilingNormalisedConfig> {
+  protected existingSections: (FormSection & { registratorId: string })[] = [];
+  protected existingRegistratorHandlers: Record<string, Pick<Parameters<RegisterSections>[0], "set" | "get">> = {};
+  protected metadata = pluginUserMetadata<{ profileConfig?: UserMetadataConfig }>(METADATA_KEY);
+
+  static ProgressiveProfilingCompletedClaim: BooleanClaim;
+
+  constructor(pluginConfig: SuperTokensPluginProfileProgressiveProfilingNormalisedConfig) {
+    super(pluginConfig);
+
+    Implementation.ProgressiveProfilingCompletedClaim = new BooleanClaim({
+      key: `${PLUGIN_ID}-completed`,
+      fetchValue: async (userId, recipeUserId, tenantId, currentPayload, userContext) => {
+        const userMetadata = await this.metadata.get(userId);
+        return this.areAllSectionsCompleted(
```

Comment:
Please test that this interacts nicely with overrides. (so `areAllSectionsCompleted` overrides actually affect what this claim does).
If it doesn't, then we may want to construct this claim in the plugin init thing directly.